MeesGame/Gameplay/GameObjects/Player.cs

```diff
@@ -129,6 +133,31 @@ public bool HasKey()
             }
             return false;
         }
+
+        public void walkAnimation(string spritesheet)
+        {
+            Stopwatch time = new Stopwatch();
+            time.Start();
+            
+            Thread animationThread = new Thread(() => {
```

Comment:
update is called all the time. Check in the update function how much game time has elasped since the previous animation. If its more then the time between animations, show the next image.